lib/Semantics/check-select-stmt.cpp

```diff
@@ -0,0 +1,398 @@
+//===-- lib/semantics/check-select-stmt.cc --------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "check-select-stmt.h"
+#include "flang/Parser/message.h"
+#include "flang/Semantics/tools.h"
+
+namespace Fortran::semantics {
+
+class SelectCaseHelper {
+
+public:
+  SelectCaseHelper(parser::Messages &messages) : messages_{messages} {}
+  void SetKind(int kind) { kind_ = kind; }
+  void SetTypeCategory(TypeCategory category) { category_ = category; }
+
+  template<typename T> std::string GetValueAsString(T val) {
+    if constexpr (std::is_same_v<T, bool>) {
+      return std::string(val ? "true" : "false");
```

Comment:
These should be ".TRUE." and ".FALSE.". But even better would be to use the folded `evaluate::Expr` and call `AsFortran` on that. Then this function wouldn't need to duplicate that functionality.

Reply:
While implimenting this, i had tired to use AsFortran interface. For this, i would have to pass the evaluated expression to multiple functions and was able to use AsFortran in just 2 places. I thought current approach would keep the code more readable. 
I have modified true/false to .TRUE./.FALSE.